staging/src/k8s.io/client-go/dynamic/fake/simple.go

```diff
@@ -116,6 +119,9 @@ func (c *dynamicResourceClient) Create(ctx context.Context, obj *unstructured.Un
 		name := accessor.GetName()
 		uncastRet, err = c.client.Fake.
 			Invokes(testing.NewCreateSubresourceAction(c.resource, name, strings.Join(subresources, "/"), c.namespace, obj), obj)
+		if err != nil {
```

Comment:
There is a check on err after the switch, why isn't that enough?

Also why some switch cases don't have this error check?

Reply:
Now that you are mentioning it ... yes, I agree.
Those additional if statements do not make any sense at all!

Reply:
Not quite! The static check noticed that err was shadowed and therefore the err was not evaluated.

Reply:
Good catch!